Catalog Storefront 2.4, Read API

mslabko opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)

Introduce changes needed for 
- magento/catalog-storefront#17
- Catalog Storefront for 2.4-develop

#### Note
PR must be force-pushed to develop-storefront  branch which currently on Magento 2.3.4. 
develop-storefront-2.4-tmp is created just for CR purpose 

<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

### Related Pull Requests
EE - https://github.com/magento/partners-magento2ee/pull/286
Catalog storefront CE - https://github.com/magento/catalog-storefront/pull/45
Catalog storefront EE - https://github.com/magento/catalog-storefront-ee/pull/5
Service connector - https://github.com/magento/magento-services-connector/pull/3
Saas export - https://github.com/magento/saas-export/pull/62
Data solutions - https://github.com/magento/data-solutions-services-id/pull/17

### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. Fixes magento/catalog-storefront#17

### Manual testing scenarios (*)
<!---
    Please provide a set of unambiguous steps to test the proposed code change.
    Giving us manual testing scenarios will help with the processing and validation process.
-->
1. ...
2. ...

### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

quangdo-aligent committed:
Ensure constructor change is backwards compatible

quangdo-aligent committed:
Suppress CookieAndSessionMisuse

quangdo-aligent committed:
Add null check for admin session user

nntoan committed:
Remove the annotations

nntoan committed:
Update fixtures

nntoan committed:
Merge branch '2.4-develop' into fallback-to-storelabels-customer-attributes

swnsma committed:
Merge branch '2.4-develop' into fallback-to-storelabels-customer-attributes

swnsma committed:
Merge branch '2.4-develop' into fallback-to-storelabels-customer-attributes

srsathish92 committed:
#27091 removed array print while setup upgrade

srsathish92 committed:
Fixed static test

Commit:
Load appropriate slider widget on demand to improve performance

quangdo-aligent committed:
Added unit test to assert cache IDs for different ACL roles must not be equal

quangdo-aligent committed:
Update class description to address static test

Commit:
Fix static tests

ihor-sviziev committed:
Load appropriate slider widget on demand to improve performance

Fix static tests

Commit:
Don't load datepicker module until it is actually needed

jiten-patel committed:
Merge branch '2.4-develop' into issue-27051

jiten-patel committed:
Merge branch '2.4-develop' into issue-27051

quangdo-aligent committed:
Update deprecated PHPUnit_Framework_MockObject_MockObject declarations

Commit:
Fix static tests

Commit:
Adjust logic to fix date formatting

Usik2203 committed:
PR 28072 Add coverage for cart items

lbajsarowicz committed:
Migrate Plugin out of Framework (to Theme module)

lbajsarowicz committed:
Fix automated checks

lbajsarowicz committed:
Fix Unit Tests location

srsathish92 committed:
Fix#27985 CMS page link active on current page

engcom-Echo committed:
coverage

ProkopovVitaliy committed:
fixed negative  children_count after deleting categories

ProkopovVitaliy committed:
Merge branch '2.4-develop' into fix-27969

ProkopovVitaliy committed:
added descriptions

ProkopovVitaliy committed:
Merge branch 'fix-27969' of github.com:ProkopovVitaliy/magento2 into fix-27969

ProkopovVitaliy committed:
Merge branch '2.4-develop' into fix-27969

engcom-Echo committed:
Cover for change

jiten-patel committed:
Merge branch '2.4-develop' into issue-27051

Commit:
Adding Wishlist GraphQl support

Usik2203 committed:
PR 28194 Add ability to update Gift Message for cart item

slavvka committed:
Merge branch '2.4-develop' into performance-datepicker

slavvka committed:
Merge branch '2.4-develop' into performance-range-slider

slavvka committed:
Merge branch '2.4-develop' into fallback-to-storelabels-customer-attributes

slavvka committed:
Merge branch '2.4-develop' into fix-27969

engcom-Echo committed:
Merge branch '2.4-develop' of http://github.com/magento/magento2 into fix/admin-category-tree-cache-key

 Conflicts:
	app/code/Magento/Catalog/Test/Unit/Ui/DataProvider/Product/Form/Modifier/CategoriesTest.php

engcom-Echo committed:
conflict resolved

p-bystritsky committed:
magento/magento2#26089: Customer Sharing Options not respected in REST API.

engcom-Echo committed:
Merge branch '2.4-develop' of http://github.com/magento/magento2 into bug/27091-remove-array-print

 Conflicts:
	setup/src/Magento/Setup/Test/Unit/Model/InstallerTest.php

engcom-Echo committed:
refactor

lbajsarowicz committed:
Merge remote-tracking branch 'origin/2.4-develop' into bugfix/move-plugin

# Conflicts:
#	app/code/Magento/Theme/Test/Unit/Plugin/LoadDesignPluginTest.php

engcom-Echo committed:
Merge branch '2.4-develop' of http://github.com/magento/magento2 into issue-27051

 Conflicts:
	app/code/Magento/Catalog/Test/Unit/Controller/Adminhtml/Product/Attribute/SaveTest.php

engcom-Echo committed:
resolved conflict

Commit:
Merge branch '2.4-develop' into feature/wishlist-graphql-258

Usik2203 committed:
Added additional checks and test cases

Usik2203 committed:
Remove redandant line from fixture

Usik2203 committed:
move part logic to other file

srsathish92 committed:
Resolved conflicts

Commit:
Test coverage for Wishlist GraphQl

Usik2203 committed:
Merge branch 'graphql-update-message-in-quote' into graphql-256-cart-item-coverage

Usik2203 committed:
changes in test fixture

Commit:
Merge branch '2.4-develop' into feature/wishlist-graphql-258

Usik2203 committed:
Added test and minor fix

Usik2203 committed:
Remove EE type from CA schema

ProkopovVitaliy committed:
Merge branch '2.4-develop' into fix-27969

lbajsarowicz committed:
Generated code is not consistent with Magento requirements and Coding Standard

Commit:
Implementing Todos and adjusting the tests

Commit:
Merge branch '2.4-develop' into feature/wishlist-graphql-258

Commit:
Fixing Unit Test

Commit:
Fixing Unit Test

ProkopovVitaliy committed:
Merge branch '2.4-develop' into fix-27969

engcom-Kilo committed:
Merge branch '2.4-develop' into fallback-to-storelabels-customer-attributes

ProkopovVitaliy committed:
Merge branch '2.4-develop' into fix-27969

kate-kyzyma committed:
Refactoring AdminGenerateUrlRewritesForProductInCategoriesSwitchOffTest

engcom-Kilo committed:
Fix Integration test.

engcom-Kilo committed:
Merge branch '2.4-develop' into fallback-to-storelabels-customer-attributes

engcom-Charlie committed:
fix issue

engcom-Charlie committed:
fix

engcom-Charlie committed:
added comment

engcom-Kilo committed:
Merge branch '2.4-develop' into performance-range-slider

VladimirZaets committed:
Merge branch '2.4-develop' into fix-27969

Commit:
Improving checking the customer wishlist

Commit:
Merge branch '2.4-develop' into feature/wishlist-graphql-258

Commit:
Fixing static tests

Usik2203 committed:
minor fix

engcom-Charlie committed:
fix

engcom-Charlie committed:
fix webapi

engcom-Charlie committed:
impr

Commit:
Merge branch '2.4-develop' into feature/wishlist-graphql-258

Commit:
Merge branch '2.4-develop' into feature/wishlist-graphql-258

litsher committed:
Changed array creation

Make the array creation consistent throughout the class

litsher committed:
Merge branch '2.4-develop' into patch-1

engcom-Echo committed:
Merge branch '2.4-develop' of http://github.com/magento/magento2 into 28357

 Conflicts:
	dev/tests/integration/testsuite/Magento/CmsUrlRewrite/Plugin/Cms/Model/Store/ViewTest.php

engcom-Echo committed:
resolved conflict

litsher committed:
Merge branch '2.4-develop' into patch-1

p-bystritsky committed:
Product should be assigned to option while option is assigned to product.

lbajsarowicz committed:
MFTF: Remove risky test that does not have implementation

Commit:
Merge branch '2.4-develop' into feature/wishlist-graphql-258

Usik2203 committed:
Add some changes for messages

Usik2203 committed:
Merge branch '2.4-develop' into graphql-256-cart-item-coverage

kate-kyzyma committed:
Merge branch '2.4-develop' into Refactoring-AdminGenerateUrlRewritesForProductInCategoriesSwitchOffTest

lenaorobei committed:
Merge branch '2.4-develop' into graphql-256-cart-item-coverage

ProkopovVitaliy committed:
Merge branch '2.4-develop' into fix-27969

Commit:
Merge branch '2.4-develop' into feature/wishlist-graphql-258

Usik2203 committed:
minor fix

Usik2203 committed:
Set gift message if gift message missing for cart item

litsher committed:
Merge branch '2.4-develop' into patch-1

kate-kyzyma committed:
Refactoring the test

Commit:
Converting the arguments in camelcase format

Commit:
Converting the arguments in camelcase format

kate-kyzyma committed:
Refactoring AdminDeleteRootCategoryTest and AdminDeleteRootSubCategoryTest

lbajsarowicz committed:
Merge branch '2.4-develop' into bugfix/move-plugin

lbajsarowicz committed:
Merge branch '2.4-develop' into bugfix/invalid-code-generated

Commit:
Merge branch '2.4-develop' into feature/wishlist-graphql-258

dmytro-ch committed:
Minor code style fix

swnsma committed:
Merge branch '2.4-develop' into fallback-to-storelabels-customer-attributes

le0n4ik committed:
[17] Merge api-proto with develop #71

nikita-shcherbatykh committed:
MC-34998: suspected degradation in performance due to targetrule

kate-kyzyma committed:
Refactoring AdminCheckSubCategoryIsNotVisibleInNavigationMenuTest

kate-kyzyma committed:
Refactoring ProductAvailableAfterEnablingSubCategoriesTest

kate-kyzyma committed:
Merge branch '2.4-develop' into Refactoring-AdminCheckSubCategoryIsNotVisibleInNavigationMenuTest

Usik2203 committed:
Merge branch '2.4-develop' into graphql-256-cart-item-coverage

Nazar65 committed:
Close prompt immediately after click ok instead of wait ajax request

engcom-Echo committed:
add fixture

skovalenk committed:
[17] Read API    Integration tests    Magento StorefrontGraphQl #63 [17] Read API    Integ

skovalenk committed:
[17] Read API :: Integration tests :: Customer* #55

mslabko committed:
Merge pull request #28746 from skovalenk/MC-PUBLIC-17-2

[17] Read API    Integration tests    Magento StorefrontGraphQl #63 […

mslabko committed:
Merge remote-tracking branch 'origin/2.4-develop' into storefront-2.4

tsviklinskyi committed:
MC-35058: Error when trying to remove categories from product

magento-engcom-team committed:
ENGCOM-7588: Generated code is not consistent with Magento requirements and Coding Standard #28351

 - Merge Pull Request magento/magento2#28351 from lbajsarowicz/magento2:bugfix/invalid-code-generated
 - Merged commits:
   1. fcc3e27f5c55d2eaa5b896bb73488c0e3a0139d8
   2. 3f87a5b2db68c341b49d5e907ab1d9ff1bdc9a38

yelahin-serhiy committed:
MC-34394: [OnPrem] QUANS - Async/bulk operations.

engcom-Echo committed:
Merge branch '2.4-develop' of http://github.com/magento/magento2 into bug/27091-remove-array-print

 Conflicts:
	setup/src/Magento/Setup/Test/Unit/Model/InstallerTest.php

engcom-Echo committed:
resolved conflict

tsviklinskyi committed:
MC-34655: [on prem] reCaptcha V3 error thrown randomly in most of the captcha enabled forms

nikita-shcherbatykh committed:
MC-34063: After Magento upgrade to 2.3.4, price "sales_order_item.price" is now incorrectly saving item value for child items in a bundle item.

nikita-shcherbatykh committed:
MC-34793: Coupon code based on shipping method remains applied after shipping method changed to another one

magento-engcom-team committed:
ENGCOM-7265: #27091 Removed array print while setup upgrade #27567

 - Merge Pull Request magento/magento2#27567 from srsathish92/magento2:bug/27091-remove-array-print
 - Merged commits:
   1. d95a65adde03f8a573b0b1dd15d0c5a247d8d99d
   2. 981df7bdc533a1209253976b2677b8af03f2df08
   3. 16ebd28965b8bda3ed32a40ea841867159cbcbfa
   4. a044c531ed9c9dc54cfc73630bbaf0ae003b7995
   5. 075f997093284c1fd1f94c0cef3c4f7b199591db
   6. 479166ef9ae0ba50c813c3f502372b05299a0c84

avattam06 committed:
Merge branch '2.4-develop' into feature/wishlist-graphql-258

magento-engcom-team committed:
ENGCOM-7420: Add ACL role ID to category tree cache id #27429

 - Merge Pull Request magento/magento2#27429 from quangdo-aligent/magento2:fix/admin-category-tree-cache-key
 - Merged commits:
   1. 54a11eb67d5ad3b0f2ca0160dcfa936d50b44d33
   2. f1c8ad56edcad49fd51288e3a0dbf2df74505df5
   3. 207a44956715bee23dbd7c0ded5735283299f9cc
   4. 61f847ec3edbf7c91ae7158d81778109c0821fbc
   5. 5a4e93f80eee1d37c63243d4e99d947aa8465a96
   6. e5b64f533b984063d9e735cf6f2b485fca6b721b
   7. 06d3e3aa19f2a51332996416a38725d03d21be05
   8. 179a72ff4736c51d6189c4422f4245dacb800004
   9. a09627037573662071d47fa0d022bd4035e45818

engcom-Echo committed:
Merge branch '2.4-develop' of http://github.com/magento/magento2 into magento/magento2#26089

nikita-shcherbatykh committed:
Merge branch '2.4-develop' into MC-34793

kate-kyzyma committed:
Merge branch '2.4-develop' into Refactoring-AdminCreateCategoryWithCustomRootCategoryTest

kate-kyzyma committed:
Updating the test

engcom-Echo committed:
exclude check expectedMessage soap

Nazar65 committed:
Introduce observers to track deleted entities

Nazar65 committed:
static test fixes

Nazar65 committed:
Satic test fixes

le0n4ik committed:
Merge pull request #28768 from magento/storefront-2.4-merge

[17] Merge api-proto with develop #71

mslabko committed:
ECP-515: Storefront Get API

- skip search tests with incorrect sort order

mslabko committed:
ECP-515: Storefront Get API

- fix tests

magento-engcom-team committed:
ENGCOM-7651: Changed array creation to be in line with the class #28515

 - Merge Pull Request magento/magento2#28515 from litsher/magento2:patch-1
 - Merged commits:
   1. 8901baa21943a9caf7fe9ea2de1f4fb5e2c82daa
   2. 4418a950dbcf5975946ae6a135eb49aef5713ff6
   3. f44321fad91df75e6631979f4a1d4257fa24fd04
   4. ba12f12f4b64e06c24fed85f6ccd368edfd94bc3

magento-engcom-team committed:
ENGCOM-7616: Product CSV import error report file missing #28460

 - Merge Pull Request magento/magento2#28460 from engcom-Charlie/magento2:28420
 - Merged commits:
   1. d4d8ef48391fdff05c9fce5d1cf013dff40d724d
   2. 597d65f0721e3de11d2e3f2c2a39e6490f90eb8b

engcom-Kilo committed:
Merge branch '2.4-develop' into performance-range-slider

magento-engcom-team committed:
ENGCOM-7602: Error when creating a store view if there are CMS pages with the same URL key #28421

 - Merge Pull Request magento/magento2#28421 from engcom-Charlie/magento2:28357
 - Merged commits:
   1. 035aaf08f8a5e5253a09bf5a7e618acf9d3e32f7
   2. f1d14503b3dd05743ce00eef444d7d248a78f6b6
   3. 9d417ec96f061e700316760c887679caab275e97

magento-engcom-team committed:
ENGCOM-7484: #27962 Migrate Plugin out of Framework (to Theme module) #27965

 - Merge Pull Request magento/magento2#27965 from lbajsarowicz/magento2:bugfix/move-plugin
 - Merged commits:
   1. d938e7bcab3f4620d5d0facb23a05ab81304e988
   2. ae58f560edad58b8c9b6fd01f9460b7c37fd2895
   3. 02aae13792fc648c00f6f8e153e208dc0d00c67d
   4. a84633e0b01bb2778418274d97d22758f534b008
   5. 7ff60630b48733141b471bc5c9eeb5c8412fc7b5

ProkopovVitaliy committed:
added a return type definition for compatibility with PHPUnit 8

Co-authored-by: Dmytro Cheshun <d.cheshun@atwix.com>

ProkopovVitaliy committed:
Merge branch '2.4-develop' into fix-27969

magento-engcom-team committed:
ENGCOM-7290: Load appropriate slider widget on demand to improve performance #27616

 - Merge Pull Request magento/magento2#27616 from krzksz/magento2:performance-range-slider
 - Merged commits:
   1. 2a90e762729951cf92aba632769a8e31a6bec851
   2. f799baa4e3f487d6f10d763baa647bf65a99ec60
   3. 62b9453becac492cafce1c0abaf0977c8afa2708
   4. 8bacfbed723fbb1342a65553cbcfe54baa36a7d9
   5. e08a0a39062d932a00b17ac9fc35bad86bfb6eb7
   6. 0e6ab8376a85580e4112a37729d4e61d436efe9b

engcom-Kilo committed:
Merge branch '2.4-develop' into fallback-to-storelabels-customer-attributes

skovalenk committed:
Fix integration test Magento\Elasticsearch6\Controller\QuickSearchTest

serhii-balko committed:
Merge remote-tracking branch 'origin/MC-35058' into 2.4-develop-pr31

serhii-balko committed:
Merge remote-tracking branch 'origin/MC-34793' into 2.4-develop-pr31

serhii-balko committed:
Merge remote-tracking branch 'origin/MC-34655' into 2.4-develop-pr31

serhii-balko committed:
Merge remote-tracking branch 'origin/MC-34063' into 2.4-develop-pr31

serhii-balko committed:
Merge remote-tracking branch 'origin/MC-34998' into 2.4-develop-pr31

serhii-balko committed:
Merge remote-tracking branch 'origin/MC-34394' into 2.4-develop-pr31

magento-engcom-team committed:
ENGCOM-7685: [GraphQl] Adding Wishlist GraphQl coverage #28205

 - Merge Pull Request magento/magento2#28205 from eduard13/magento2:feature/wishlist-graphql-258
 - Merged commits:
   1. 5de8d9b20fc44d80b2ea1eced24a9e68e9439010
   2. eddb4ebbc475ada3adf8081975f86b1aad89d3f8
   3. 7488a36001628ec19b364ad08f7886f65b172da9
   4. 3346f33df01d98decea3e3d8257fef0ecf637d68
   5. 28b1d020546d8e3b1dd8297236b199686d8767fa
   6. c988ab26e875d0cd9794300e6adfe676780896d1
   7. 58215aafb22108ecc30337a87ccf80fd1ecb0270
   8. 73650e3c4259dfe1c2bb29daa29d55cbc6462682
   9. a5cc7903b1eeda59df56bb3683d2823fc7c880ad
   10. 9e023a42eada7c3903d98403ed89abcd6f0bcc3a
   11. 08bd47efff266a548c4544062ed7544ca1ecde4b
   12. 27c3142925a92de2a6edc9094ee5e22f7450f25e
   13. 71ebd52761eb8c92313f6098f5ef8a835ce18af5
   14. 83708180e1014c2574578af5bea43d8f634d8124
   15. f0a4bc953fa7992cf7199f6e320c34a02ff8ce87
   16. 476ea33acaaaff969ef41837ed68f1c4b2090d2f
   17. f1d13b7dc2177e4eeb5b776e438d76fcd7df9d32
   18. 5bc151bb10237c64f3b32419356cb322229077c8
   19. b71144d8aed73d1d54403ac13b463b4f43e5fca5

magento-engcom-team committed:
ENGCOM-7244: Customer attribute frontend labels not using the store label values on the frontend #27064

 - Merge Pull Request magento/magento2#27064 from nntoan/magento2:fallback-to-storelabels-customer-attributes
 - Merged commits:
   1. fdec82c179e1e3de4505d2ad62321ce303e56ff3
   2. 9a7d68d176e1300dea2d52f41357cf79d71bec4c
   3. 0bf66d9f82f793c9ec205f38e0f9005c2233297b
   4. 9b6e0586eb97211e8e713f3d680ad459fbb8ce32
   5. aa148c330e63db3917cb89f5ee192bb710218207
   6. 4c110b7cbaf65db19112d8b030240801bf2ee22a
   7. 705b5be7119c31284ebf862bdf0a9b6bf2a909d7
   8. 5749b24d5090354c2f5fd3d6569b1297b3810a1b
   9. d4ba184dbd9d371713ac998dbdb00077a0f725cc
   10. cec82daeb82b39cc5643e5a14eadad9dd5119cd1
   11. 2f0dae98faa6b1ec47162516965d7f2b1a28af1e
   12. 88da24aea8f0d535324a7196c09fe37c78bbd08a
   13. e99e6e170146da3b97cc63d43e4d0a3dea7e5c64
   14. 5bc8bba7e68e71767b77b1fdb0a3d39d9f8638ae
   15. 9f20b7f030ca5c9a399a3ec556fd391a1fa306a0
   16. b1480fbb4e127ffd9e36e7a78d1b380a9b90321a
   17. 05fbd1ff95b6f6f1093068482b0f1e96b906d634
   18. ef806a5447d438558bcd604e083e8cbef96590d6
   19. dec273fc98927ca8ffb97595f907322aecb3a371
   20. 458ce8d9c511c734f078a0ecc6130db913ce9575
   21. 8848c6aa9cea99ab1b9c164d5ca517522c9a9e68
   22. 53498e1721b69223184182c6beb883d7393d05b6
   23. 893641cbfe37b1f272f3b3954d74d663f39c38c4
   24. 48411f918dc42acbd6f860b2465177b3897cd8fe
   25. b4f98d603029b69e5cebf2f5a9a87751f76e4257
   26. 040c23d1b6e47f5d5b37e69636f97899f0d33e6e
   27. fc9b8419198f96914fb64ed679ecffe6afa1cadb
   28. 6a912c2a42a1184250923f7376f1b247c7d22f95
   29. e9b4cf0a7cc4e726bf5c3c24b76f63edce321f09

magento-engcom-team committed:
ENGCOM-7513: fixed negative  children_count after deleting categories #28044

 - Merge Pull Request magento/magento2#28044 from ProkopovVitaliy/magento2:fix-27969
 - Merged commits:
   1. 0dfd146b819b9bfd0ca0cd72e41e8c56c3debcae
   2. 0fea657d349e9f509ee980cc2e1eabcdc41c7f84
   3. fc65f5b3ca97449c21c29ca4bf38b2b48d776577
   4. ce3d97f86e8a858e1877c8888eacf6e21f603c19
   5. e7b6e452f74b9178a15ceac2b68f2904aace6a45
   6. b07a831bf8b47876f2624349272e4cd6029ee03b
   7. 85ecd6edfd0790312ff192ec08378c754b45074d
   8. fd7663f9d14b1e7e22f35f3be795d7dce22a5bfc
   9. 0242849f73e1c4b9728832ecd44ff467b307ea25
   10. 1ff938c3e1603810f2650f502a3405dded07203b
   11. caf9ae2112cf37db04cce4a545e7cca8c7f6c151
   12. 80e3107caaaf4173ffebeeedb92ec179ca47010e
   13. 5300e3eb24bd11ae687974937078fb7976630fb2
   14. 62ee4a28c4d32eb5d428492753c856d3cc71879f

mslabko committed:
Merge pull request #28826 from skovalenk/MC-PUBLIC-102

Fix integration test Magento\Elasticsearch6\Controller\QuickSearchTest

magento-engcom-team committed:
ENGCOM-7559: [WIP][GraphQl] Gift message attribute coverage for cart item #28072

 - Merge Pull Request magento/magento2#28072 from Usik2203/magento2:graphql-256-cart-item-coverage
 - Merged commits:
   1. 034af404deaa448c6ae660166912b0c40dfc7b12
   2. 7e6e0239c9895982b4bb1577412b0018beef1361
   3. 81737c084b0b80b48cf5b3359e3b145c64aa54f1
   4. 120a105b5adb6fbc4d35a0067d72aa1975acccc5
   5. 035c1577d0e5dee7c102375cd2bdba0f1542c85f
   6. 8ed5018fda262de3fb337fc886daaede98122f96
   7. cea930b4d7549bcf476e621ad11ffbfa4847e0e6
   8. dabbd3c07dbc5ee4c09b785f3f315b4fd3c29fa1
   9. f6ce31d43aa8316ff59821c508d7b54c734bcee9
   10. a2f28d83e8ceab42e60505af371d3e5905ee1d65
   11. 233b695c4f55b1a4c14009ecc13ced4658d5b868
   12. d127c98a404e3d5e393e14157cd71b41b0df8ae7
   13. d9748d2d6eecb98b6eca36b5aa9b7602b6bf237f
   14. d994276cbab26239732d64b8c313a38eac13f01c
   15. 938e43bfae3774afde30523951aa9edd27778ec3
   16. d339ae8ec16a96edcaf6927376a3a0c179826a76

magento-engcom-team committed:
ENGCOM-7690: Close prompt immediately after click ok instead of wait ajax request #28739

 - Merge Pull Request magento/magento2#28739 from Nazar65/magento2:ASI-840
 - Merged commits:
   1. 2ac53f86438231d800aff5e2a2fa4f586063d0fb

mslabko committed:
Merge remote-tracking branch 'mainline/2.4-develop' into storefront-2.4

prabhuram93 committed:
ENGCOM-7685: [GraphQl] Adding Wishlist GraphQl coverage #28205

prabhuram93 committed:
ENGCOM-7559: [WIP][GraphQl] Gift message attribute coverage for cart item #28072

Commit:
Update schema.graphqls

Adjusting schema's descriptions

Commit:
Update schema.graphqls

Adjusting schema's descriptions

Commit:
Small schema fix

Commit:
Small schema fix

zakdma committed:
Merge pull request #5833 from magento-tsg-csl3/2.4-develop-pr31

[TSG-CSL3] For 2.4 (pr31)

Commit:
Small schema fixes

Commit:
Small schema fixes

magento-engcom-team committed:
ENGCOM-7665: magento/magento2#26089: Customer Sharing Options not respected in REST API. #28213

 - Merge Pull Request magento/magento2#28213 from p-bystritsky/magento2:magento/magento2#26089
 - Merged commits:
   1. ddee80419bf117ba64a73ae5a1e4e872e9cbe925
   2. c17449a4b835b9a2b3f63bd14d22a7294049e8c5
   3. c09e59b1f8c909379b64076da26db5c011285bf4
   4. d21ded32be1c597a2547eae3ffb5a5d18a43f9d7
   5. ce359e054eb18249e461a22ee0c954898a96b398

magento-engcom-team committed:
ENGCOM-7635: MFTF: Remove risky test that does not have implementation #28605

 - Merge Pull Request magento/magento2#28605 from lbajsarowicz/magento2:feature/remove-risky-test
 - Merged commits:
   1. f27bf4419a6176e71d75ffa72b81be5ea0e0d86e

magento-engcom-team committed:
ENGCOM-7459: Don't load datepicker module until it is actually needed #27860

 - Merge Pull Request magento/magento2#27860 from krzksz/magento2:performance-datepicker
 - Merged commits:
   1. 4c21e460d180de2d76810cddd6a53d256113656c
   2. 20c5b5c1b5963485b2b02ed66a89b74e2b3965fe
   3. e1caa4b333b3b07eb352e1d95ad431fd82924931
   4. f86d92b2f2af8e21027b41e0d51115b748a95fe1

magento-engcom-team committed:
ENGCOM-7709: Introduce observers to track deleted entities #28798

 - Merge Pull Request magento/magento2#28798 from Nazar65/magento2:ASI-delete-observers
 - Merged commits:
   1. a2d51131606d59e3124919f120243cfd69d998f3
   2. 3bc82395fede9a07a51464ff21c7b467befbdd2e
   3. 34bff1266d63294251bb96798f2c7cafcaba8122

mslabko committed:
ECP-515: Storefront Get API

- skip test with wrong sort order

magento-engcom-team committed:
ENGCOM-7514: Fix #27985 CMS page link active on current page #28004

 - Merge Pull Request magento/magento2#28004 from srsathish92/magento2:bug/27985-cms-page-link-active-issue
 - Merged commits:
   1. d5cc8cb120da0072dee54c7cc850241319b9c2c1
   2. 331ce388f1cf5f24d8667081e37641928d0c159e

magento-engcom-team committed:
[Magento Community Engineering] Community Contributions - 2.4-develop

 - merged latest code from mainline branch

slavvka committed:
ENGCOM-7709: Introduce observers to track deleted entities #28798

slavvka committed:
ENGCOM-7665: magento/magento2#26089: Customer Sharing Options not respected in REST API. #28213

slavvka committed:
ENGCOM-7635: MFTF: Remove risky test that does not have implementation #28605

slavvka committed:
ENGCOM-7514: Fix #27985 CMS page link active on current page #28004

slavvka committed:
ENGCOM-7459: Don't load datepicker module until it is actually needed #27860

mslabko committed:
ECP-515: Storefront Get API

- fix tests

le0n4ik committed:
ECP-515: Storefront Get API

    - fix tests

le0n4ik committed:
Merge branch 'storefront-2.4' of github.com:magento/magento2 into storefront-2.4

slavvka committed:
ENGCOM-7513: fixed negative  children_count after deleting categories #28044

slavvka committed:
ENGCOM-7290: Load appropriate slider widget on demand to improve performance #27616

slavvka committed:
ENGCOM-7244: Customer attribute frontend labels not using the store label values on the frontend #27064

prabhuram93 committed:
Merge branch 'feature/wishlist-graphql-258' of github.com:eduard13/magento2 into 2.4-develop-temporary-one-prs

prabhuram93 committed:
Merge branch '2.4-develop-temporary-one-prs' of github.com:magento-engcom/magento2ce into 2.4-develop-temporary-one-prs

magento-engcom-team committed:
:arrows_clockwise: [GraphQL] Community Contributions - 2.4-develop expedited-1

Accepted Community Pull Requests:
 - magento/magento2#28205: [GraphQl] Adding Wishlist GraphQl coverage (by @eduard13)
 - magento/magento2#28072: [GraphQl] Gift Message coverage for cart item (by @Usik2203)


Fixed GitHub Issues:
 - magento/magento2#246: GD2 not working in WYSIWYG (reported by @barbazul) has been fixed in magento/magento2#28072 by @Usik2203 in 2.4-develop branch
   Related commits:
     1. 034af404deaa448c6ae660166912b0c40dfc7b12
     2. 7e6e0239c9895982b4bb1577412b0018beef1361
     3. 81737c084b0b80b48cf5b3359e3b145c64aa54f1
     4. 120a105b5adb6fbc4d35a0067d72aa1975acccc5
     5. 035c1577d0e5dee7c102375cd2bdba0f1542c85f
     6. 8ed5018fda262de3fb337fc886daaede98122f96
     7. cea930b4d7549bcf476e621ad11ffbfa4847e0e6
     8. dabbd3c07dbc5ee4c09b785f3f315b4fd3c29fa1
     9. f6ce31d43aa8316ff59821c508d7b54c734bcee9
     10. a2f28d83e8ceab42e60505af371d3e5905ee1d65
     11. 233b695c4f55b1a4c14009ecc13ced4658d5b868
     12. d127c98a404e3d5e393e14157cd71b41b0df8ae7
     13. d9748d2d6eecb98b6eca36b5aa9b7602b6bf237f
     14. d994276cbab26239732d64b8c313a38eac13f01c
     15. 938e43bfae3774afde30523951aa9edd27778ec3
     16. d339ae8ec16a96edcaf6927376a3a0c179826a76

 - magento/magento2#253: PHP Extension 0 must be loaded - XML Parse mistake? (reported by @ScreamingDev) has been fixed in magento/magento2#28072 by @Usik2203 in 2.4-develop branch
   Related commits:
     1. 034af404deaa448c6ae660166912b0c40dfc7b12
     2. 7e6e0239c9895982b4bb1577412b0018beef1361
     3. 81737c084b0b80b48cf5b3359e3b145c64aa54f1
     4. 120a105b5adb6fbc4d35a0067d72aa1975acccc5
     5. 035c1577d0e5dee7c102375cd2bdba0f1542c85f
     6. 8ed5018fda262de3fb337fc886daaede98122f96
     7. cea930b4d7549bcf476e621ad11ffbfa4847e0e6
     8. dabbd3c07dbc5ee4c09b785f3f315b4fd3c29fa1
     9. f6ce31d43aa8316ff59821c508d7b54c734bcee9
     10. a2f28d83e8ceab42e60505af371d3e5905ee1d65
     11. 233b695c4f55b1a4c14009ecc13ced4658d5b868
     12. d127c98a404e3d5e393e14157cd71b41b0df8ae7
     13. d9748d2d6eecb98b6eca36b5aa9b7602b6bf237f
     14. d994276cbab26239732d64b8c313a38eac13f01c
     15. 938e43bfae3774afde30523951aa9edd27778ec3
     16. d339ae8ec16a96edcaf6927376a3a0c179826a76

 - magento/magento2#28519: GraphQL Checkout :: Add support for Gift Wrapping / Gift Message during checkout. (reported by @nrkapoor) has been fixed in magento/magento2#28072 by @Usik2203 in 2.4-develop branch
   Related commits:
     1. 034af404deaa448c6ae660166912b0c40dfc7b12
     2. 7e6e0239c9895982b4bb1577412b0018beef1361
     3. 81737c084b0b80b48cf5b3359e3b145c64aa54f1
     4. 120a105b5adb6fbc4d35a0067d72aa1975acccc5
     5. 035c1577d0e5dee7c102375cd2bdba0f1542c85f
     6. 8ed5018fda262de3fb337fc886daaede98122f96
     7. cea930b4d7549bcf476e621ad11ffbfa4847e0e6
     8. dabbd3c07dbc5ee4c09b785f3f315b4fd3c29fa1
     9. f6ce31d43aa8316ff59821c508d7b54c734bcee9
     10. a2f28d83e8ceab42e60505af371d3e5905ee1d65
     11. 233b695c4f55b1a4c14009ecc13ced4658d5b868
     12. d127c98a404e3d5e393e14157cd71b41b0df8ae7
     13. d9748d2d6eecb98b6eca36b5aa9b7602b6bf237f
     14. d994276cbab26239732d64b8c313a38eac13f01c
     15. 938e43bfae3774afde30523951aa9edd27778ec3
     16. d339ae8ec16a96edcaf6927376a3a0c179826a76

magento-engcom-team committed:
[Magento Community Engineering] Community Contributions - 2.4-develop

 - merged latest code from mainline branch

slavvka committed:
Merge branch '2.4-develop' into Refactoring-AdminCreateCategoryWithCustomRootCategoryTest

slavvka committed:
Merge branch '2.4-develop' into Refactoring-AdminGenerateUrlRewritesForProductInCategoriesSwitchOffTest

slavvka committed:
ENGCOM-7651: Changed array creation to be in line with the class #28515

slavvka committed:
ENGCOM-7616: Product CSV import error report file missing #28460

slavvka committed:
ENGCOM-7602: Error when creating a store view if there are CMS pages with the same URL key #28421

slavvka committed:
ENGCOM-7484: #27962 Migrate Plugin out of Framework (to Theme module) #27965

slavvka committed:
Merge branch '2.4-develop' into 22431

slavvka committed:
ENGCOM-7690: Close prompt immediately after click ok instead of wait ajax request #28739

slavvka committed:
ENGCOM-7588: Generated code is not consistent with Magento requirements and Coding Standard #28351

slavvka committed:
ENGCOM-7420: Add ACL role ID to category tree cache id #27429

slavvka committed:
ENGCOM-7265: #27091 Removed array print while setup upgrade #27567

magento-engcom-team committed:
:arrows_clockwise: [Magento Community Engineering] Community Contributions - 2.4-develop daily delivery

Accepted Community Pull Requests:
 - magento/magento2#28044: fixed negative  children_count after deleting categories (by @ProkopovVitaliy)
 - magento/magento2#27616: Load appropriate slider widget on demand to improve performance (by @krzksz)
 - magento/magento2#27064: Customer attribute frontend labels not using the store label values on the frontend (by @nntoan)


Fixed GitHub Issues:
 - magento/magento2#27969: children_count can become negative after deleting categories (reported by @wojtekn) has been fixed in magento/magento2#28044 by @ProkopovVitaliy in 2.4-develop branch
   Related commits:
     1. 0dfd146b819b9bfd0ca0cd72e41e8c56c3debcae
     2. 0fea657d349e9f509ee980cc2e1eabcdc41c7f84
     3. fc65f5b3ca97449c21c29ca4bf38b2b48d776577
     4. ce3d97f86e8a858e1877c8888eacf6e21f603c19
     5. e7b6e452f74b9178a15ceac2b68f2904aace6a45
     6. b07a831bf8b47876f2624349272e4cd6029ee03b
     7. 85ecd6edfd0790312ff192ec08378c754b45074d
     8. fd7663f9d14b1e7e22f35f3be795d7dce22a5bfc
     9. 0242849f73e1c4b9728832ecd44ff467b307ea25
     10. 1ff938c3e1603810f2650f502a3405dded07203b
     11. caf9ae2112cf37db04cce4a545e7cca8c7f6c151
     12. 80e3107caaaf4173ffebeeedb92ec179ca47010e
     13. 5300e3eb24bd11ae687974937078fb7976630fb2
     14. 62ee4a28c4d32eb5d428492753c856d3cc71879f

 - magento/magento2#28807: [Issue] Load appropriate slider widget on demand to improve performance (reported by @m2-assistant[bot]) has been fixed in magento/magento2#27616 by @krzksz in 2.4-develop branch
   Related commits:
     1. 2a90e762729951cf92aba632769a8e31a6bec851
     2. f799baa4e3f487d6f10d763baa647bf65a99ec60
     3. 62b9453becac492cafce1c0abaf0977c8afa2708
     4. 8bacfbed723fbb1342a65553cbcfe54baa36a7d9
     5. e08a0a39062d932a00b17ac9fc35bad86bfb6eb7
     6. 0e6ab8376a85580e4112a37729d4e61d436efe9b

 - magento/magento2#27063: [2.3.x] Customer attribute frontend labels not using the store label values on the frontend (reported by @nntoan) has been fixed in magento/magento2#27064 by @nntoan in 2.4-develop branch
   Related commits:
     1. fdec82c179e1e3de4505d2ad62321ce303e56ff3
     2. 9a7d68d176e1300dea2d52f41357cf79d71bec4c
     3. 0bf66d9f82f793c9ec205f38e0f9005c2233297b
     4. 9b6e0586eb97211e8e713f3d680ad459fbb8ce32
     5. aa148c330e63db3917cb89f5ee192bb710218207
     6. 4c110b7cbaf65db19112d8b030240801bf2ee22a
     7. 705b5be7119c31284ebf862bdf0a9b6bf2a909d7
     8. 5749b24d5090354c2f5fd3d6569b1297b3810a1b
     9. d4ba184dbd9d371713ac998dbdb00077a0f725cc
     10. cec82daeb82b39cc5643e5a14eadad9dd5119cd1
     11. 2f0dae98faa6b1ec47162516965d7f2b1a28af1e
     12. 88da24aea8f0d535324a7196c09fe37c78bbd08a
     13. e99e6e170146da3b97cc63d43e4d0a3dea7e5c64
     14. 5bc8bba7e68e71767b77b1fdb0a3d39d9f8638ae
     15. 9f20b7f030ca5c9a399a3ec556fd391a1fa306a0
     16. b1480fbb4e127ffd9e36e7a78d1b380a9b90321a
     17. 05fbd1ff95b6f6f1093068482b0f1e96b906d634
     18. ef806a5447d438558bcd604e083e8cbef96590d6
     19. dec273fc98927ca8ffb97595f907322aecb3a371
     20. 458ce8d9c511c734f078a0ecc6130db913ce9575
     21. 8848c6aa9cea99ab1b9c164d5ca517522c9a9e68
     22. 53498e1721b69223184182c6beb883d7393d05b6
     23. 893641cbfe37b1f272f3b3954d74d663f39c38c4
     24. 48411f918dc42acbd6f860b2465177b3897cd8fe
     25. b4f98d603029b69e5cebf2f5a9a87751f76e4257
     26. 040c23d1b6e47f5d5b37e69636f97899f0d33e6e
     27. fc9b8419198f96914fb64ed679ecffe6afa1cadb
     28. 6a912c2a42a1184250923f7376f1b247c7d22f95
     29. e9b4cf0a7cc4e726bf5c3c24b76f63edce321f09

magento-engcom-team committed:
[Magento Community Engineering] Community Contributions - 2.4-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[Magento Community Engineering] Community Contributions - 2.4-develop

 - merged latest code from mainline branch

le0n4ik committed:
ECP-515: Storefront Get API

    - fix tests

magento-engcom-team committed:
[Magento Community Engineering] Community Contributions - 2.4-develop-prs

 - merged with '2.4-develop-fast-lane-prs' branch

magento-engcom-team committed:
ENGCOM-7620: MFTF: Refactoring AdminGenerateUrlRewritesForProductInCategoriesSwitchOffTest #28401

 - Merge Pull Request magento/magento2#28401 from kate-kyzyma/magento2:Refactoring-AdminGenerateUrlRewritesForProductInCategoriesSwitchOffTest
 - Merged commits:
   1. 75a7d850627c9fb134b264cff30d6e99e1cd74d3
   2. d532c83faf6f686f9b9611aaabc87f2fc197432e
   3. 8d664bf57caaf6fb11362ec6268e78d0e722d0e1

magento-engcom-team committed:
ENGCOM-7712: MFTF: Refactoring the test AdminCreateCategoryWithCustomRootCategoryTest #28679

 - Merge Pull Request magento/magento2#28679 from kate-kyzyma/magento2:Refactoring-AdminCreateCategoryWithCustomRootCategoryTest
 - Merged commits:
   1. c503af964a85aa5d57f2fe7cf26cf06074f7362c
   2. ce0943160b5f60cd31a105fd33db8df86b677254
   3. f4341f8477b1f02a92e412172b31a0d0e9fe30e4
   4. dc6342ea881cfa32699cf07ee62c44e6eee011be

slavvka committed:
ENGCOM-7712: MFTF: Refactoring the test AdminCreateCategoryWithCustomRootCategoryTest #28679

slavvka committed:
ENGCOM-7620: MFTF: Refactoring AdminGenerateUrlRewritesForProductInCategoriesSwitchOffTest #28401

magento-engcom-team committed:
:arrows_clockwise: [Magento Community Engineering] Community Contributions - 2.4-develop expedited

Accepted Community Pull Requests:
 - magento/magento2#28798: Introduce observers to track deleted entities (by @Nazar65)
 - magento/magento2#28213: magento/magento2#26089: Customer Sharing Options not respected in REST API. (by @p-bystritsky)
 - magento/magento2#28605: MFTF: Remove risky test that does not have implementation (by @lbajsarowicz)
 - magento/magento2#28004: Fix #27985 CMS page link active on current page (by @srsathish92)
 - magento/magento2#27860: Don't load datepicker module until it is actually needed (by @krzksz)


Fixed GitHub Issues:
 - magento/magento2#1474: Rest Api gives HTTP 404 (reported by @sanuj) has been fixed in magento/magento2#28798 by @Nazar65 in 2.4-develop branch
   Related commits:
     1. a2d51131606d59e3124919f120243cfd69d998f3
     2. 3bc82395fede9a07a51464ff21c7b467befbdd2e
     3. 34bff1266d63294251bb96798f2c7cafcaba8122

 - magento/magento2#26089: Customer Sharing Options not respected in REST API (reported by @cebesoft) has been fixed in magento/magento2#28213 by @p-bystritsky in 2.4-develop branch
   Related commits:
     1. ddee80419bf117ba64a73ae5a1e4e872e9cbe925
     2. c17449a4b835b9a2b3f63bd14d22a7294049e8c5
     3. c09e59b1f8c909379b64076da26db5c011285bf4
     4. d21ded32be1c597a2547eae3ffb5a5d18a43f9d7
     5. ce359e054eb18249e461a22ee0c954898a96b398

 - magento/magento2#28850: [Issue] MFTF: Remove risky test that does not have implementation (reported by @m2-assistant[bot]) has been fixed in magento/magento2#28605 by @lbajsarowicz in 2.4-develop branch
   Related commits:
     1. f27bf4419a6176e71d75ffa72b81be5ea0e0d86e

 - magento/magento2#27985: Privacy and Cookie Policy link not display as active on current page. (reported by @RHapani) has been fixed in magento/magento2#28004 by @srsathish92 in 2.4-develop branch
   Related commits:
     1. d5cc8cb120da0072dee54c7cc850241319b9c2c1
     2. 331ce388f1cf5f24d8667081e37641928d0c159e

 - magento/magento2#28823: [Issue] Don't load datepicker module until it is actually needed (reported by @m2-assistant[bot]) has been fixed in magento/magento2#27860 by @krzksz in 2.4-develop branch
   Related commits:
     1. 4c21e460d180de2d76810cddd6a53d256113656c
     2. 20c5b5c1b5963485b2b02ed66a89b74e2b3965fe
     3. e1caa4b333b3b07eb352e1d95ad431fd82924931
     4. f86d92b2f2af8e21027b41e0d51115b748a95fe1

magento-engcom-team committed:
[Magento Community Engineering] Community Contributions - 2.4-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
[Magento Community Engineering] Community Contributions - 2.4-develop

 - merged latest code from mainline branch

magento-engcom-team committed:
:arrows_clockwise: [Magento Community Engineering] Community Contributions - 2.4-develop

Accepted Community Pull Requests:
 - magento/magento2#28739: Close prompt immediately after click ok instead of wait ajax request (by @Nazar65)
 - magento/magento2#28351: Generated code is not consistent with Magento requirements and Coding Standard (by @lbajsarowicz)
 - magento/magento2#27429: Add ACL role ID to category tree cache id (by @quangdo-aligent)
 - magento/magento2#27567: #27091 Removed array print while setup upgrade (by @srsathish92)
 - magento/magento2#28515: Changed array creation to be in line with the class (by @litsher)
 - magento/magento2#28460: Product CSV import error report file missing (by @engcom-Charlie)
 - magento/magento2#28421: Error when creating a store view if there are CMS pages with the same URL key (by @engcom-Charlie)
 - magento/magento2#27965: #27962 Migrate Plugin out of Framework (to Theme module) (by @lbajsarowicz)


Fixed GitHub Issues:
 - magento/magento2#28376: [Issue] Generated code is not consistent with Magento requirements and Coding Standard (reported by @m2-backlog[bot]) has been fixed in magento/magento2#28351 by @lbajsarowicz in 2.4-develop branch
   Related commits:
     1. fcc3e27f5c55d2eaa5b896bb73488c0e3a0139d8
     2. 3f87a5b2db68c341b49d5e907ab1d9ff1bdc9a38

 - magento/magento2#28306: [Issue] Add ACL role ID to category tree cache id (reported by @m2-backlog[bot]) has been fixed in magento/magento2#27429 by @quangdo-aligent in 2.4-develop branch
   Related commits:
     1. 54a11eb67d5ad3b0f2ca0160dcfa936d50b44d33
     2. f1c8ad56edcad49fd51288e3a0dbf2df74505df5
     3. 207a44956715bee23dbd7c0ded5735283299f9cc
     4. 61f847ec3edbf7c91ae7158d81778109c0821fbc
     5. 5a4e93f80eee1d37c63243d4e99d947aa8465a96
     6. e5b64f533b984063d9e735cf6f2b485fca6b721b
     7. 06d3e3aa19f2a51332996416a38725d03d21be05
     8. 179a72ff4736c51d6189c4422f4245dacb800004
     9. a09627037573662071d47fa0d022bd4035e45818

 - magento/magento2#27091: Magento 2.3.4 array print while setup upgrade (reported by @RHapani) has been fixed in magento/magento2#27567 by @srsathish92 in 2.4-develop branch
   Related commits:
     1. d95a65adde03f8a573b0b1dd15d0c5a247d8d99d
     2. 981df7bdc533a1209253976b2677b8af03f2df08
     3. 16ebd28965b8bda3ed32a40ea841867159cbcbfa
     4. a044c531ed9c9dc54cfc73630bbaf0ae003b7995
     5. 075f997093284c1fd1f94c0cef3c4f7b199591db
     6. 479166ef9ae0ba50c813c3f502372b05299a0c84

 - magento/magento2#28795: [Issue] Changed array creation to be in line with the class (reported by @m2-assistant[bot]) has been fixed in magento/magento2#28515 by @litsher in 2.4-develop branch
   Related commits:
     1. 8901baa21943a9caf7fe9ea2de1f4fb5e2c82daa
     2. 4418a950dbcf5975946ae6a135eb49aef5713ff6
     3. f44321fad91df75e6631979f4a1d4257fa24fd04
     4. ba12f12f4b64e06c24fed85f6ccd368edfd94bc3

 - magento/magento2#28420: Product CSV import error report file missing (reported by @mischabraam) has been fixed in magento/magento2#28460 by @engcom-Charlie in 2.4-develop branch
   Related commits:
     1. d4d8ef48391fdff05c9fce5d1cf013dff40d724d
     2. 597d65f0721e3de11d2e3f2c2a39e6490f90eb8b

 - magento/magento2#28357: Error when creating a store view if there are CMS pages with the same URL key (reported by @stupachenko) has been fixed in magento/magento2#28421 by @engcom-Charlie in 2.4-develop branch
   Related commits:
     1. 035aaf08f8a5e5253a09bf5a7e618acf9d3e32f7
     2. f1d14503b3dd05743ce00eef444d7d248a78f6b6
     3. 9d417ec96f061e700316760c887679caab275e97

 - magento/magento2#27962: Plugins in \Magento\Framework namespace (reported by @lbajsarowicz) has been fixed in magento/magento2#27965 by @lbajsarowicz in 2.4-develop branch
   Related commits:
     1. d938e7bcab3f4620d5d0facb23a05ab81304e988
     2. ae58f560edad58b8c9b6fd01f9460b7c37fd2895
     3. 02aae13792fc648c00f6f8e153e208dc0d00c67d
     4. a84633e0b01bb2778418274d97d22758f534b008
     5. 7ff60630b48733141b471bc5c9eeb5c8412fc7b5

magento-engcom-team committed:
ENGCOM-7511: Fixed issue 27051: Saving an attribute with backend_type static #27369

 - Merge Pull Request magento/magento2#27369 from jiten-patel/magento2:issue-27051
 - Merged commits:
   1. b06db1f34b6799255a1cc7f596e21190ebaec7f2
   2. aaa7054cd1e679bcba32c2311a3e547a52c2d590
   3. fa154cde355e41e19f9e9a677807c71ae1d535be
   4. 7d8b893bf3f86c5d1467125fc091b1692b9b4201
   5. 054bf5d80633b40ba16a264a334be09510bbbe55
   6. 2160c372b988fc1f09863fbc178613b28db3842f
   7. 83545c04624316805cce50a97b141f367d6605ec
   8. c1a01fc4c797e037c760285b9434df52c4a38373
   9. eeb7433fe287c0d9200a3836aa54fcba88a7f11e
   10. 7d2d0d3f369176264d12215f58ee6d5f3b067d62
   11. eb343417c7fbe5b0692646686457e150d7d52fae

magento-engcom-team committed:
ENGCOM-7676: MFTF: Refactoring AdminDeleteRootCategoryTest and AdminDeleteRootSubCategor… #28684

 - Merge Pull Request magento/magento2#28684 from kate-kyzyma/magento2:Refactoring-AdminDeleteCategoryTest
 - Merged commits:
   1. b1849f6d4ff6ee1289e3065e1bddbba54b3f0471
   2. f09c981cd4c086e9981ee3810392a824f8cdf0ad

magento-engcom-team committed:
ENGCOM-7708: MFTF: Replace repetitive actions with Action Groups for AdminCheckSubCategoryIsNotVisibleInNavigationMenuTest and ProductAvailableAfterEnablingSubCategoriesTest #28727

 - Merge Pull Request magento/magento2#28727 from kate-kyzyma/magento2:Refactoring-AdminCheckSubCategoryIsNotVisibleInNavigationMenuTest
 - Merged commits:
   1. 834959491135ec26a4d7cb2f4b9d3602f063481d
   2. 7a6e6465495684a9a08bb5a5c03afb5b6515143e
   3. d358ace02f50feae2c01e1441a93eeae864ef81d

le0n4ik committed:
ECP-515: Storefront Get API

    - fix tests

magento-engcom-team committed:
[Magento Community Engineering] Community Contributions - 2.4-develop

 - merged latest code from mainline branch

slavvka committed:
ENGCOM-7708: MFTF: Replace repetitive actions with Action Groups for AdminCheckSubCategoryIsNotVisibleInNavigationMenuTest and ProductAvailableAfterEnablingSubCategoriesTest #28727

slavvka committed:
ENGCOM-7676: MFTF: Refactoring AdminDeleteRootCategoryTest and AdminDeleteRootSubCategor… #28684

slavvka committed:
Merge branch '2.4-develop' into 22431

slavvka committed:
ENGCOM-7511: Fixed issue 27051: Saving an attribute with backend_type static #27369

vzabaznov committed:
[90] [17] Read API :: Web-Api tests :: Magento\GraphQl\Catalog\MediaGalleryTest::testProductSmallImageUrlPlaceholder #90

vzabaznov committed:
[90] [17] Read API :: Web-Api tests :: Magento\GraphQl\Catalog\MediaGalleryTest::testProductSmallImageUrlPlaceholder #90

mslabko commented on dev/tests/integration/framework/Magento/TestFramework/Db/DynamicTables.php:
```diff
@@ -0,0 +1,38 @@
+     */
+    public function createTables()
+    {
+        $this->categoryProductIndexTables->createTables();
```
@akaplya Do we have task to handle this in more abstract way so theses changes can be merged to 2.4-develop?

magento-engcom-team committed:
:arrows_clockwise: [Magento Community Engineering] Community Contributions - 2.4-develop latest changes

Accepted Community Pull Requests:
 - magento/magento2#28679: MFTF: Refactoring the test AdminCreateCategoryWithCustomRootCategoryTest (by @kate-kyzyma)
 - magento/magento2#28401: MFTF: Refactoring AdminGenerateUrlRewritesForProductInCategoriesSwitchOffTest (by @kate-kyzyma)
 - magento/magento2#28727: MFTF: Replace repetitive actions with Action Groups for AdminCheckSubCategoryIsNotVisibleInNavigationMenuTest and ProductAvailableAfterEnablingSubCategoriesTest (by @kate-kyzyma)
 - magento/magento2#28684: MFTF: Refactoring AdminDeleteRootCategoryTest and AdminDeleteRootSubCategor… (by @kate-kyzyma)

mslabko commented on dev/tests/integration/framework/Magento/TestFramework/Db/DymanicTables/CategoryProductIndexTables.php:
```diff
@@ -0,0 +1,53 @@
+    public function createTables(): void
+    {
+        $connection = $this->resourceConnection->getConnection();
+        for ($storeId = 0; $storeId <= 256; $storeId++) {
```
What test we fix with this ad-hoc solution?

melnikovi replied:
As I remember it's for dev/tests/integration/testsuite/Magento/CatalogImportExport/Model/ProductTest.php. This code is in mainline already.

mslabko commented on dev/tests/integration/testsuite/Magento/Framework/MessageQueue/UseCase/WaitAndNotWaitMessagesTest.php:
```diff
@@ -66,14 +66,21 @@ protected function setUp(): void
+        $this->publisherConsumerController->stopConsumers();
+
+        $config = $this->config;
+        $config['queue']['consumers_wait_for_messages'] = 1;
```
@le0n4ik Looks like we need to roll back to an original config value otherwise we broke the isolation

le0n4ik replied:
@mslabko actually it's rolling back in 
\Magento\Framework\MessageQueue\UseCase\WaitAndNotWaitMessagesTest::tearDown

Config will be written from $this->config variable which was stored in setUp() function

le0n4ik replied:
> @le0n4ik Looks like we need to roll back to an original config value otherwise we broke the isolation

@mslabko actually it's rolling back in
\Magento\Framework\MessageQueue\UseCase\WaitAndNotWaitMessagesTest::tearDown

Config will be written from $this->config variable which was stored in setUp() function

mslabko commented on dev/tests/integration/testsuite/Magento/Wishlist/Model/WishlistTest.php:
```diff
@@ -116,17 +116,13 @@ public function testGetItemCollection(): void
     {
-        $productSku = 'simple';
         $customerId = 1;
-        $product = $this->productRepository->get($productSku);
```
@skovalenk  here we have another test:
Originally we tested, that product, that was added to Wishlist will not be available to Customer if it was disabled

Now, we test, that disabled product added to wishlist will not be available to Customer. It's a different cases

vzabaznov replied:
I've changed wishlist_with_disabled_simple_product.php fixture in order to keep previous behaviour

magento-engcom-team committed:
[Magento Community Engineering] Community Contributions - 2.4-develop

 - merged latest code from mainline branch

buskamuza commented on dev/tests/api-functional/testsuite/Magento/GraphQl/Catalog/ProductViewTest.php:
```diff
@@ -665,7 +667,8 @@ private function assertMediaGalleryEntries($product, $actualResponse)
         $this->assertCount(1, $mediaGalleryEntries, "Precondition failed, incorrect number of media gallery entries.");
-        $this->assertIsArray([$actualResponse['media_gallery_entries']],
+        $this->assertIsArray(
+            [$actualResponse['media_gallery_entries']],
```
This is strange. It is always array.

magento-engcom-team committed:
:arrows_clockwise: [Magento Community Engineering] Community Contributions - 2.4-develop

Accepted Community Pull Requests:
 - magento/magento2#27369: Fixed issue 27051: Saving an attribute with backend_type static (by @jiten-patel)


Fixed GitHub Issues:
 - magento/magento2#27051: Saving an attribute with backend_type 'static' will remove the content of the 'frontend_class' field. (reported by @MLucaciu) has been fixed in magento/magento2#27369 by @jiten-patel in 2.4-develop branch
   Related commits:
     1. b06db1f34b6799255a1cc7f596e21190ebaec7f2
     2. aaa7054cd1e679bcba32c2311a3e547a52c2d590
     3. fa154cde355e41e19f9e9a677807c71ae1d535be
     4. 7d8b893bf3f86c5d1467125fc091b1692b9b4201
     5. 054bf5d80633b40ba16a264a334be09510bbbe55
     6. 2160c372b988fc1f09863fbc178613b28db3842f
     7. 83545c04624316805cce50a97b141f367d6605ec
     8. c1a01fc4c797e037c760285b9434df52c4a38373
     9. eeb7433fe287c0d9200a3836aa54fcba88a7f11e
     10. 7d2d0d3f369176264d12215f58ee6d5f3b067d62
     11. eb343417c7fbe5b0692646686457e150d7d52fae

magento-engcom-team committed:
ENGCOM-7612: Rest API Magento 2.4 order update overwrites existing item information #28483

 - Merge Pull Request magento/magento2#28483 from engcom-Charlie/magento2:22431
 - Merged commits:
   1. a922551cedec33ab3f831590fa5c5a97ff90d470
   2. beb41d03ca35b6a423f546f31149df57124475bb
   3. 3115fe9aca02e3de26163921493bf7a103ea62fe
   4. 15d20cb39f5143aed71f10b7310c38a180f50a14
   5. ae36ae4cc8f07d8573acf63ed01082715f00c433

slavvka committed:
ENGCOM-7612: Rest API Magento 2.4 order update overwrites existing item information #28483

magento-engcom-team committed:
:arrows_clockwise: [Magento Community Engineering] Community Contributions - 2.4-develop

Accepted Community Pull Requests:
 - magento/magento2#28483: Rest API Magento 2.4 order update overwrites existing item information (by @engcom-Charlie)


Fixed GitHub Issues:
 - magento/magento2#22431: Rest API Magento 2.4 order update overwrites existing item information (reported by @aschrammel) has been fixed in magento/magento2#28483 by @engcom-Charlie in 2.4-develop branch
   Related commits:
     1. a922551cedec33ab3f831590fa5c5a97ff90d470
     2. beb41d03ca35b6a423f546f31149df57124475bb
     3. 3115fe9aca02e3de26163921493bf7a103ea62fe
     4. 15d20cb39f5143aed71f10b7310c38a180f50a14
     5. ae36ae4cc8f07d8573acf63ed01082715f00c433

le0n4ik committed:
Merge remote-tracking branch 'origin/storefront-2.4' into storefront-2.4

le0n4ik committed:
ECP-515: Storefront Get API

    - fix tests

le0n4ik committed:
Merge branch 'storefront-2.4' of github.com:magento/magento2 into storefront-2.4

mslabko committed:
ECP-515: Storefront Get API

mslabko committed:
ECP-515: Storefront Get API

le0n4ik committed:
Merge remote-tracking branch 'origin/2.4-develop' into storefront-2.4

vzabaznov committed:
[90] [17] Read API :: Web-Api tests :: Magento\GraphQl\Catalog\MediaGalleryTest::testProductSmallImageUrlPlaceholder #90

le0n4ik committed:
Revert "Merge remote-tracking branch 'origin/2.4-develop' into storefront-2.4"

This reverts commit de71581f4906e85b0b52645a783ef8c5e96af0b1, reversing
changes made to aa71b85bac25c0fbaf0bf276efdd7bbb3e213d29.

mslabko committed:
ECP-515: Storefront Get API

- fix test

le0n4ik committed:
ECP-515: Storefront Get API

    - fix static tests
